src/Loggers/OwnFormattingChannel.cpp

```diff
@@ -1,18 +1,19 @@
 #include "OwnFormattingChannel.h"
+#include "OwnJSONPatternFormatter.h"
 #include "OwnPatternFormatter.h"
-
-
 namespace DB
 {
-
 void OwnFormattingChannel::logExtended(const ExtendedLogMessage & msg)
 {
     if (pChannel && priority >= msg.base.getPriority())
     {
-        if (pFormatter)
+        if (pFormatterJSON || pFormatter)
         {
             std::string text;
-            pFormatter->formatExtended(msg, text);
+            if (pFormatterJSON)
```

Comment:
There is no need to store JSON formatter in a separate field, separately handle it, and even include it. I suggest removing all changes from OwnFormattingChannel.cpp and OwnFormattingChannel.h

Reply:
I tried that, but that doesn't call appropriate format function to print correct output. However, will check again.

Reply:
Maybe overriding `formatExtended()` in `src/Loggers/OwnJSONPatternFormatter.h` will help?

As a second option we could use something like this:
```
if (auto * formatter = dynamic_cast<DB:: OwnJSONPatternFormatter *>(pFormatter.get()))
```

Reply:
Applied the second approach and committed. Thanks for the suggestion :)